WalletWasabi.Fluent/Models/UI/ApplicationSettings.cs

```diff
@@ -0,0 +1,214 @@
+
+		_config = config;
+		_uiConfig = uiConfig;
+
+		// Advanced
+		_enableGpu = _config.EnableGpu;
+
+		// Bitcoin
+		_network = _config.Network;
+		_startLocalBitcoinCoreOnStartup = _config.StartLocalBitcoinCoreOnStartup;
+		_localBitcoinCoreDataDir = _config.LocalBitcoinCoreDataDir;
+		_stopLocalBitcoinCoreOnShutdown = _config.StopLocalBitcoinCoreOnShutdown;
+		_bitcoinP2PEndPoint = _config.GetBitcoinP2pEndPoint().ToString(defaultPort: -1);
+		_dustThreshold = _config.DustThreshold.ToString();
+
+		// General
+		_darkModeEnabled = _uiConfig.DarkModeEnabled;
+		_autoCopy = _uiConfig.Autocopy;
+		_autoPaste = _uiConfig.AutoPaste;
+		_customChangeAddress = _uiConfig.IsCustomChangeAddress;
+		_selectedFeeDisplayUnit = Enum.IsDefined(typeof(FeeDisplayUnit), _uiConfig.FeeDisplayUnit)
+			? (FeeDisplayUnit)_uiConfig.FeeDisplayUnit
+			: FeeDisplayUnit.Satoshis;
+		_runOnSystemStartup = _uiConfig.RunOnSystemStartup;
+		_hideOnClose = _uiConfig.HideOnClose;
+		_useTor = _config.UseTor;
+		_terminateTorOnExit = _config.TerminateTorOnExit;
+		_downloadNewVersion = _config.DownloadNewVersion;
+
+		// Privacy Mode
+		_privacyMode = _uiConfig.PrivacyMode;
+
+		// Save on change
+		this.WhenAnyValue(
+			x => x.EnableGpu,
+			x => x.Network,
+			x => x.StartLocalBitcoinCoreOnStartup,
+			x => x.LocalBitcoinCoreDataDir,
+			x => x.StopLocalBitcoinCoreOnShutdown,
+			x => x.BitcoinP2PEndPoint,
+			x => x.DustThreshold,
+			x => x.UseTor,
+			x => x.TerminateTorOnExit,
+			x => x.DownloadNewVersion)
+			.Skip(1)
+			.ObserveOn(RxApp.MainThreadScheduler)
+			.Throttle(TimeSpan.FromMilliseconds(ThrottleTime))
+			.Do(_ => Save())
+			.Subscribe();
+
+		// Save UiConfig on change
+		this.WhenAnyValue(
+			x => x.DarkModeEnabled,
+			x => x.AutoCopy,
+			x => x.AutoPaste,
+			x => x.CustomChangeAddress,
+			x => x.SelectedFeeDisplayUnit,
+			x => x.RunOnSystemStartup,
+			x => x.HideOnClose,
+			x => x.PrivacyMode)
+			.Skip(1)
+			.Throttle(TimeSpan.FromMilliseconds(ThrottleTime))
+			.Do(_ => ApplyUiConfigChanges())
+			.Subscribe();
+
+		// Set Default BitcoinCoreDataDir if required
+		this.WhenAnyValue(x => x.StartLocalBitcoinCoreOnStartup)
+			.Skip(1)
+			.Where(value => value && string.IsNullOrEmpty(LocalBitcoinCoreDataDir))
+			.Subscribe(_ => LocalBitcoinCoreDataDir = EnvironmentHelpers.GetDefaultBitcoinCoreDataDirOrEmptyString());
+
+		// Apply RunOnSystenStartup
+		this.WhenAnyValue(x => x.RunOnSystemStartup)
+			.DoAsync(async x => await StartupHelper.ModifyStartupSettingAsync(RunOnSystemStartup))
+			.Subscribe();
+	}
+
+	public IObservable<bool> IsRestartNeeded => _isRestartNeeded;
+
+	public bool CheckIfRestartIsNeeded(PersistentConfig config)
+	{
+		return !_startupConfig.AreDeepEqual(config);
+	}
+
+	private void Save()
+	{
+		var config = new PersistentConfig(_startupConfig.FilePath);
+
+		RxApp.MainThreadScheduler.Schedule(
+			() =>
+			{
+				try
+				{
+					lock (ConfigLock)
+					{
+						// TODO: Roland: do we really need to do this?
```

Comment:
Yes, otherwise the config would contain the default values instead of the ones that are stored on the disk.